apps/frontend/src/pages/project/libs/components/project-details-menu/project-details-menu.tsx

```diff
@@ -1,36 +1,51 @@
-import { useNavigate } from "react-router-dom";
-
-import { Menu, MenuItem } from "~/libs/components/components.js";
-import { AppRoute } from "~/libs/enums/enums.js";
-import { configureString } from "~/libs/helpers/helpers.js";
-import { useCallback, usePopover } from "~/libs/hooks/hooks.js";
+import { Menu, MenuItem, NavLink } from "~/libs/components/components.js";
+import { AppRoute, PermissionKey } from "~/libs/enums/enums.js";
+import { checkHasPermission, configureString } from "~/libs/helpers/helpers.js";
+import { useAppSelector, useCallback, usePopover } from "~/libs/hooks/hooks.js";
 
 type Properties = {
+	onEdit: () => void;
 	projectId: number;
 };
 
-const ProjectDetailsMenu = ({ projectId }: Properties): JSX.Element => {
+const ProjectDetailsMenu = ({
+	onEdit,
+	projectId,
+}: Properties): JSX.Element | null => {
+	const { isOpened, onClose, onOpen } = usePopover();
+	const { authenticatedUser } = useAppSelector(({ auth }) => auth);
+
+	const handleEditClick = useCallback(() => {
+		onEdit();
+		onClose();
+	}, [onEdit, onClose]);
+
+	if (!authenticatedUser) {
+		return null;
+	}
+
 	const projectAccessManagementRoute = configureString(
 		AppRoute.PROJECT_ACCESS_MANAGEMENT,
 		{
 			id: projectId.toString(),
 		},
 	);
 
-	const { isOpened, onClose, onOpen } = usePopover();
-	const navigate = useNavigate();
+	const hasManageAllProjectsPermission = checkHasPermission(
```

Comment:
we will have multiple permissions here
```suggestion
	const hasManageAccessPermission = checkHasPermission(
```

Reply:
Changed the naming